Update rebar3 plugins

michalwski opened the pull request:
This PR addresses #

Proposed changes include:
* describe the functionality changes
* describe new or updated tests
* describe changes to the documentation

michalwski committed:
Update rebar3 plugins

mongoose-im commented:
[1616.1](https://travis-ci.org/michalwski/MongooseIM/jobs/691376552) / Erlang 22.0 / small_tests / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-26_16.02.45/lib.mongooseim.logs/run.2020-05-26_16.02.45/suite.log.html)

---

[1616.2](https://travis-ci.org/michalwski/MongooseIM/jobs/691376553) / Erlang 22.0 / internal_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/internal_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/internal_mnesia.22.0/big/ct_run.test%40travis-job-8b9a8cff-b65b-4139-8d0e-b358f4f51108.2020-05-26_16.04.20/index.html)
**OK: 1460** / Failed: 0 / **User-skipped: 157** / Auto-skipped: 0 

---

[1616.4](https://travis-ci.org/michalwski/MongooseIM/jobs/691376555) / Erlang 22.0 / mysql_redis / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/mysql_redis.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/mysql_redis.22.0/big/ct_run.test%40travis-job-01e73458-afb3-47fa-b228-e55e7bf70a48.2020-05-26_16.05.07/index.html)
**OK: 2728** / Failed: 0 / **User-skipped: 217** / Auto-skipped: 0 

---

[1616.3](https://travis-ci.org/michalwski/MongooseIM/jobs/691376554) / Erlang 22.0 / odbc_mssql_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/odbc_mssql_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/odbc_mssql_mnesia.22.0/big/ct_run.test%40travis-job-f5ea01b3-2e67-4274-a79c-fa5dc908d0cf.2020-05-26_16.04.59/index.html)
**OK: 2733** / Failed: 0 / **User-skipped: 212** / Auto-skipped: 0 

---

[1616.5](https://travis-ci.org/michalwski/MongooseIM/jobs/691376556) / Erlang 22.0 / riak_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/riak_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/riak_mnesia.22.0/big/ct_run.test%40travis-job-76d0ea7f-0593-42ee-ba7d-67dab4e2b8d3.2020-05-26_16.05.34/index.html)
**OK: 1586** / Failed: 0 / **User-skipped: 177** / Auto-skipped: 0 

---

[1616.7](https://travis-ci.org/michalwski/MongooseIM/jobs/691376559) / Erlang 22.0 / elasticsearch_and_cassandra_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/elasticsearch_and_cassandra_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/elasticsearch_and_cassandra_mnesia.22.0/big/ct_run.test%40travis-job-b21107b0-3ba9-4bba-ad18-80bba35ca43a.2020-05-26_16.26.14/index.html)
**OK: 330** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0 

---

[1616.6](https://travis-ci.org/michalwski/MongooseIM/jobs/691376557) / Erlang 22.0 / ldap_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/ldap_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/ldap_mnesia.22.0/big/ct_run.test%40travis-job-7e37e9ce-9199-4b26-a518-e771fa9063f1.2020-05-26_16.16.47/index.html)
**OK: 1368** / Failed: 0 / **User-skipped: 249** / Auto-skipped: 0 

---

[1616.9](https://travis-ci.org/michalwski/MongooseIM/jobs/691376561) / Erlang 21.3 / pgsql_mnesia / f28b5292b672a54a9818c828e78397fcdd24dac8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1616/pgsql_mnesia.21.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/pgsql_mnesia.21.3/big/ct_run.test%40travis-job-1c29b70a-58c9-4b0d-8b96-e4241725fbd7.2020-05-26_16.33.31/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1616/pgsql_mnesia.21.3/small/ct_run.mongooseim@localhost.2020-05-26_16.28.42/lib.mongooseim.logs/run.2020-05-26_16.28.43/suite.log.html)
**OK: 2746** / Failed: 0 / **User-skipped: 199** / Auto-skipped: 0

michalwski committed:
Update rebar3 to 3.13.2

mongoose-im commented:
[1617.1](https://travis-ci.org/michalwski/MongooseIM/jobs/691439768) / Erlang 22.0 / small_tests / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-26_18.59.49/lib.mongooseim.logs/run.2020-05-26_18.59.50/suite.log.html)

---

[1617.2](https://travis-ci.org/michalwski/MongooseIM/jobs/691439769) / Erlang 22.0 / internal_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/internal_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/internal_mnesia.22.0/big/ct_run.test%40travis-job-d9b19fe4-5051-406e-8815-3dc7b2a0e041.2020-05-26_19.01.52/index.html)
**OK: 1460** / Failed: 0 / **User-skipped: 157** / Auto-skipped: 0 

---

[1617.4](https://travis-ci.org/michalwski/MongooseIM/jobs/691439771) / Erlang 22.0 / mysql_redis / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/mysql_redis.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/mysql_redis.22.0/big/ct_run.test%40travis-job-d7b94de7-82c2-4cbe-ba33-f403f80e101d.2020-05-26_19.02.09/index.html)
**OK: 2728** / Failed: 0 / **User-skipped: 217** / Auto-skipped: 0 

---

[1617.3](https://travis-ci.org/michalwski/MongooseIM/jobs/691439770) / Erlang 22.0 / odbc_mssql_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/odbc_mssql_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/odbc_mssql_mnesia.22.0/big/ct_run.test%40travis-job-09c5e331-7a8c-4daf-a44a-34a130637969.2020-05-26_19.02.30/index.html)
**OK: 2733** / Failed: 0 / **User-skipped: 212** / Auto-skipped: 0 

---

[1617.5](https://travis-ci.org/michalwski/MongooseIM/jobs/691439772) / Erlang 22.0 / riak_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/riak_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/riak_mnesia.22.0/big/ct_run.test%40travis-job-599d897c-2522-430f-963e-1f39c884114d.2020-05-26_19.02.29/index.html)
**OK: 1586** / Failed: 0 / **User-skipped: 177** / Auto-skipped: 0 

---

[1617.6](https://travis-ci.org/michalwski/MongooseIM/jobs/691439773) / Erlang 22.0 / ldap_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/ldap_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/ldap_mnesia.22.0/big/ct_run.test%40travis-job-adfe2d2a-8796-4022-88f6-378e0a058f88.2020-05-26_19.12.06/index.html)
**OK: 1368** / Failed: 0 / **User-skipped: 249** / Auto-skipped: 0 

---

[1617.7](https://travis-ci.org/michalwski/MongooseIM/jobs/691439774) / Erlang 22.0 / elasticsearch_and_cassandra_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/elasticsearch_and_cassandra_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/elasticsearch_and_cassandra_mnesia.22.0/big/ct_run.test%40travis-job-e5c0b7dd-08e2-477d-a96d-0345a7ac7380.2020-05-26_19.24.03/index.html)
**OK: 330** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0 

---

[1617.9](https://travis-ci.org/michalwski/MongooseIM/jobs/691439776) / Erlang 21.3 / pgsql_mnesia / 070a33e7aa369d9107a02bb89db31c7175f3a299
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1617/pgsql_mnesia.21.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/pgsql_mnesia.21.3/big/ct_run.test%40travis-job-c7760fd3-8f06-418d-b23d-3e6874cf80fa.2020-05-26_19.31.40/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1617/pgsql_mnesia.21.3/small/ct_run.mongooseim@localhost.2020-05-26_19.26.53/lib.mongooseim.logs/run.2020-05-26_19.26.53/suite.log.html)
**OK: 2746** / Failed: 0 / **User-skipped: 199** / Auto-skipped: 0

michalwski committed:
Disable rebar_faster_deps because it crashes rebar3

This pluging is currently outdate and doesn't work with newest rebar3.
We need newer rebar3 as with the older one (3.6) I observed issues
downloading some hex packages.

According to the following warning:
```
===> [rebar_faster_deps] Example: PREFER_GIT_CLONE=true ./rebar3 get-deps
===> Using custom resource rebar_gitcache_resource that implements a deprecated api. It should be upgraded to rebar_resource_v2.
```

This plugin uses deprecated api, which most probably crashes newest
rebar3 when upgrading some dependencies.

michalwski committed:
use esl/rebar3_codecov plugin

mongoose-im commented:
[1621.1](https://travis-ci.org/michalwski/MongooseIM/jobs/691699148) / Erlang 22.0 / small_tests / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-27_11.19.51/lib.mongooseim.logs/run.2020-05-27_11.19.52/suite.log.html)

---

[1621.2](https://travis-ci.org/michalwski/MongooseIM/jobs/691699149) / Erlang 22.0 / internal_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/internal_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/internal_mnesia.22.0/big/ct_run.test%40travis-job-1b20b5d6-e30b-4db4-8629-98321c6bfabb.2020-05-27_11.21.03/index.html)
**OK: 1460** / Failed: 0 / **User-skipped: 157** / Auto-skipped: 0 

---

[1621.4](https://travis-ci.org/michalwski/MongooseIM/jobs/691699151) / Erlang 22.0 / mysql_redis / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/mysql_redis.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/mysql_redis.22.0/big/ct_run.test%40travis-job-87f186de-9f43-4a0c-b593-232300297485.2020-05-27_11.24.44/index.html)
**OK: 2728** / Failed: 0 / **User-skipped: 217** / Auto-skipped: 0 

---

[1621.3](https://travis-ci.org/michalwski/MongooseIM/jobs/691699150) / Erlang 22.0 / odbc_mssql_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/odbc_mssql_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/odbc_mssql_mnesia.22.0/big/ct_run.test%40travis-job-820a3f48-5b75-4543-8f93-39e9dfefdd93.2020-05-27_11.24.39/index.html)
**OK: 2733** / Failed: 0 / **User-skipped: 212** / Auto-skipped: 0 

---

[1621.7](https://travis-ci.org/michalwski/MongooseIM/jobs/691699154) / Erlang 22.0 / elasticsearch_and_cassandra_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/elasticsearch_and_cassandra_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/elasticsearch_and_cassandra_mnesia.22.0/big/ct_run.test%40travis-job-e99f210e-d4bf-4253-bcf0-736caa01f08b.2020-05-27_11.42.40/index.html)
**OK: 330** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0 

---

[1621.6](https://travis-ci.org/michalwski/MongooseIM/jobs/691699153) / Erlang 22.0 / ldap_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/ldap_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/ldap_mnesia.22.0/big/ct_run.test%40travis-job-d8e73c77-adae-4c6b-9fb5-1abe169ad401.2020-05-27_11.38.40/index.html)
**OK: 1368** / Failed: 0 / **User-skipped: 249** / Auto-skipped: 0 

---

[1621.5](https://travis-ci.org/michalwski/MongooseIM/jobs/691699152) / Erlang 22.0 / riak_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/riak_mnesia.22.0)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/riak_mnesia.22.0/big/ct_run.test%40travis-job-cbb222b7-4525-4b91-8d98-ede0e8205eb6.2020-05-27_11.33.19/index.html)
**OK: 1586** / Failed: 0 / **User-skipped: 177** / Auto-skipped: 0 

---

[1621.9](https://travis-ci.org/michalwski/MongooseIM/jobs/691699156) / Erlang 21.3 / pgsql_mnesia / 516fca4f646ff7012b7759d9398a5293eae4335b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/175/1621/pgsql_mnesia.21.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/pgsql_mnesia.21.3/big/ct_run.test%40travis-job-38f09dc9-951b-487c-b1bd-ed4462c67566.2020-05-27_11.53.49/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/175/1621/pgsql_mnesia.21.3/small/ct_run.mongooseim@localhost.2020-05-27_11.49.04/lib.mongooseim.logs/run.2020-05-27_11.49.05/suite.log.html)
**OK: 2746** / Failed: 0 / **User-skipped: 199** / Auto-skipped: 0
